server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByWithoutValueFilterDataSet.java

```diff
@@ -56,22 +55,13 @@
 
   protected Map<PartialPath, GroupByExecutor> pathExecutors = new HashMap<>();
```

Comment:
Since you have have already split all the process of aligned and non-align path, you can replace the `pathExecutors ` to two `Map` like `resultIndexes` and `alignedPathToIndexesMap`. If so, you don't have to cast them to corresponding exact class each time.

Reply:
fixed